### PR TITLE
add a command line option to only run a specific case id

### PR DESCRIPTION
**Pull request by madina0801**
Issue #203
I just didn't run lint script and got too many warnings. I still get some though, but at least not as many 😄
I cannot get the filtering part, where and how I should filter main tasks, and changed configuration in "main.ts" file. I'm open to any kind of feedback! Thank you in advance :)

### CODE REVIEW

File: framework/src/build-task-list.ts [hunk 1]
```diff
@@ -5,14 +5,16 @@ import {
...
   for (const task of tasks) {
-    if (task.pre) {
-      task.pre = task.pre.filter((entry,) => ! blacklist.includes(entry,),);
-    }
-    if (task.post) {
-      task.post = task.post.filter((entry,) => ! blacklist.includes(entry,),);
-    }
-    for (let i=0; i<total; i ++) {
-      internalTasks.push(task,);
+    if (task.main) {
```
Comment (Idrinth) [inline framework/src/build-task-list.ts:8]: `task.main` will always be set, but this is where you could do the filtering by checking `task.id`

File: framework/bin/run-benchmark.js [hunk 2]
```diff
@@ -1,4 +1,16 @@
 #!/usr/bin/env node
-import cli from '../src/cli/bench.js';
+import {
+  cli, run,
+} from '../src/cli/bench.js';
 
 cli(process.argv,);
```
Comment (Idrinth) [inline framework/bin/run-benchmark.js:6]: this already runs the process. move the argument parsing into cli :)

File: framework/src/main.ts [hunk 3]
```diff
@@ -42,7 +42,8 @@ export const run = async(
     progress?: Progress,
     language?: string,
     blacklist?: string[],
-    mode?: 'benchmarking'|'content-testing'|'load-testing'
+    mode?: 'benchmarking'|'content-testing'|'load-testing',
+    taskId?: string,
```
Comment (Idrinth) [inline framework/src/main.ts:46]: this would need to be handed through to the executor and it's task builder

**Comment by Idrinth** [comment]
> I just didn't run lint script and got too many warnings. I still get some though, but at least not as many 😄

Don't worry too much about it, the linter is set to very strict, so it's rare to not find something

> I cannot get the filtering part, where and how I should filter main tasks, and changed configuration in "main.ts" file.

You will have to hand through the id to run from main to executor to the task list builder. In that you can filter by checking `task.id` against the taskId you provided.

> I'm open to any kind of feedback! Thank you in advance :)

Thanks for the effort, this looks very good as a starting point!

File: framework/src/build-task-list.ts (hunk 1 above)
Comment (madina0801) [reply framework/src/build-task-list.ts:8]: Should it look something like this? 
```
if (taskId && task.id !== taskId) {
      continue;
    }
```

File: framework/bin/run-benchmark.js (hunk 2 above)
Comment (madina0801) [reply framework/bin/run-benchmark.js:6]: Oh, okay I see it now, sorry :) Can I use args array and constants that are already provided?

File: framework/src/main.ts (hunk 3 above)
Comment (madina0801) [reply framework/src/main.ts:46]: I handed it through to the executor function and passed taskId to the BuildTaskList as the second parameter. Is it right?

File: framework/bin/run-benchmark.js (hunk 2 above)
Comment (Idrinth) [reply framework/bin/run-benchmark.js:6]: of course

File: framework/src/main.ts (hunk 3 above)
Comment (Idrinth) [reply framework/src/main.ts:46]: sounds very good so far

File: framework/src/build-task-list.ts (hunk 1 above)
Comment (Idrinth) [reply framework/src/build-task-list.ts:8]: no continue, otherwise looks good. Feel free to extract it into a small function

**Comment by madina0801** [comment]
Could you please kindly review?^^ And maybe reach out to me on Discord: @madss20 ? Cause it's getting quite long :) have a nice rest of your day!

**Comment by Idrinth** [comment]
@madina0801 sent you a friend request on discord, username should be obvious :)
I believe I underestimated this task :( I didn't expect it to be this complicated.

**Pull request merged**